internal/kafka/internal/handlers/cluster.go

```diff
@@ -86,6 +75,22 @@ func (h clusterHandler) RegisterEnterpriseCluster(w http.ResponseWriter, r *http
 				return nil, errors.New(errors.ErrorUnauthorized, "non admin user not authorized to perform this action")
 			}
 
+			ownershipCheckErr := provider.CheckIfOrganizationIsTheClusterOwner(orgId, clusterPayload.ClusterId, clusterSpec.ExternalID)
+			if ownershipCheckErr != nil {
+				serviceErr, ok := ownershipCheckErr.(*errors.ServiceError)
+				if !ok {
+					return nil, errors.NewWithCause(errors.ErrorGeneral, ownershipCheckErr, ownershipCheckErr.Error())
+				}
+				return nil, serviceErr
```

Comment:
I'd prefer the opposite (`if ok` instead of `if !ok`):

``` go
	serviceErr, ok := ownershipCheckErr.(*errors.ServiceError)
	if ok {
		return nil, serviceErr
	}
	return nil, errors.NewWithCause(errors.ErrorGeneral, ownershipCheckErr, ownershipCheckErr.Error())
```

or

```go
	if serviceErr, ok := ownershipCheckErr.(*errors.ServiceError); ok {
		return nil, serviceErr	
	}
	return nil, errors.NewWithCause(errors.ErrorGeneral, ownershipCheckErr, ownershipCheckErr.Error())
```

Reply:
+1  I had forgotten to refactor it.